op-challenger/game/fault/preimages/large.go

```diff
@@ -128,11 +128,13 @@ func (p *LargePreimageUploader) initLargePreimage(ctx context.Context, uuid *big
 func (p *LargePreimageUploader) addLargePreimageLeafs(ctx context.Context, uuid *big.Int, chunks []Chunk) error {
 	queue := txmgr.NewQueue[int](ctx, p.txMgr, 10)
 	receiptChs := make([]chan txmgr.TxReceipt[int], len(chunks))
+	blocksProcessed := int64(0)
 	for i, chunk := range chunks {
-		tx, err := p.contract.AddLeaves(uuid, chunk.Input, chunk.Commitments, chunk.Finalize)
+		tx, err := p.contract.AddLeaves(uuid, big.NewInt(blocksProcessed), chunk.Input, chunk.Commitments, chunk.Finalize)
 		if err != nil {
 			return fmt.Errorf("failed to create pre-image oracle tx: %w", err)
 		}
+		blocksProcessed += int64(len(chunk.Input) / matrix.LeafSize)
```

Comment:
I'd move this onto the `Chunk` type itself, ie `Chunk.blocks() int64` that calculates the number of leaves in the chunk. We can then unit test that method and then it is not hidden deep inside the `addLargePreimageLeafs` method

Reply:
https://github.com/ethereum-optimism/optimism/pull/9143 moves all this into `StateMatrix` anyway and it will be able to count the individual leaves it adds easily without this division.  So I'd probably just land this as-is.